mlos_bench/db/db_schema_pg.sql

```diff
@@ -0,0 +1,106 @@
+--
+-- Copyright (c) Microsoft Corporation.
+-- Licensed under the MIT License.
+--
+-- PostgreSQL version of the DB schema.
+
+DROP TABLE IF EXISTS experiment_config CASCADE;
+DROP TABLE IF EXISTS experiment_merge CASCADE;
+DROP TABLE IF EXISTS trial_status CASCADE;
+DROP TABLE IF EXISTS trial_config CASCADE;
+DROP TABLE IF EXISTS trial_results CASCADE;
+DROP TABLE IF EXISTS trial_telemetry CASCADE;
+
+DROP TYPE IF EXISTS experiment_id_t CASCADE;
+DROP TYPE IF EXISTS trial_id_t CASCADE;
+DROP TYPE IF EXISTS param_id_t CASCADE;
+DROP TYPE IF EXISTS status_t CASCADE;
+
+CREATE DOMAIN experiment_id_t AS VARCHAR(128) NOT NULL;
+CREATE DOMAIN trial_id_t AS INTEGER NOT NULL;
+CREATE DOMAIN param_id_t AS VARCHAR(256) NOT NULL;
+
+-- Should match the `mlos_bench.environment.Status` enum.
+CREATE TYPE status_t AS ENUM (
+    'UNKNOWN',
+    'PENDING',
+    'READY',
+    'RUNNING',
+    'SUCCEEDED',
+    'CANCELED',
+    'FAILED',
+    'TIMED_OUT'
+);
+
+CREATE TABLE experiment_config (
+    exp_id experiment_id_t,
+    descr TEXT,
+    git_repo VARCHAR(255) NOT NULL,
+    git_commit VARCHAR(40) NOT NULL,
+
+    PRIMARY KEY (exp_id)
+);
+
+CREATE SEQUENCE trial_id_seq;
+
+CREATE TABLE trial_status (
+    exp_id experiment_id_t,
+    trial_id trial_id_t DEFAULT nextval('trial_id_seq'),
```

Comment:
Was dealing with this in benchbase just now, so I'll post here as well:
https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-identity-column/
Postgres added support for a *slightly* more standard way of doing "auto-increment" with `GENERATED ALWAYS AS IDENTITY` syntax.
That's probably a good thing for us to use going forward since it's in the SQL spec there's some hope that other DBs can use it as is as well.

Reply:
I was thinking about it, and I've decided not to use auto-increment at all, regardless of the DB engine. We actually need a trial id as a sequence starting with 1 for each experiment. Otherwise, the numbering of trials gets very confusing. Also, less variation in syntax across different DBs